internal/handler/user.go

```diff
@@ -0,0 +1,90 @@
+package handler
+
+import (
+	"errors"
+	userModel "github.com/Azzonya/gophermart/internal/domain/user"
+	"github.com/Azzonya/gophermart/internal/storage"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type Jsn struct {
+	Host         string `json:"host"`
+	ShortMessage string `json:"short_message"`
+	Err          string `json:"_err"`
+}
+
+func (u *UserHandlers) RegisterUser(c *gin.Context) {
+	var err error
+
+	req := &userModel.User{}
+
+	ctx := c.Request.Context()
+
+	err = c.BindJSON(req)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Failed to read body",
+			"error":   err.Error(),
+		})
+		return
+	}
+
+	newUser, err := u.userUsecase.Register(ctx, req)
```

Comment:
Это логическое продолжение комментария выше, который в импортах - есть импорт и сервиса и инжектируется usecase (что правильно). Нужно решить какой допускается доступ и зависимость в хэндлерах. Т.е. четкое правило - все хэндлеры зависят исключительно от usecase. Тогда эта строчка верная. А иначе что мешает сделать вот так?
```suggestion
	newUser, err := u.UserModel.Register(ctx, req)
```